tests/phpunit/testcases/core/functions.php

```diff
@@ -878,59 +878,85 @@ public function override_initial_version() {
 	}
 
 	/**
+	 * This test acts like a reminder: $deprecated_functions_versions needs to include each versions of the `/src/bp-core/deprecated` directory.
+	 *
+	 * eg:  if `/src/bp-core/deprecated/15.0.php` exists, then $deprecated_functions_versions should include `15.0`.
+	 *
+	 * @group bp_get_deprecated_functions_versions
+	 * @ticket BP9210
+	 */
+	public function test_bp_get_deprecated_functions_versions_array_is_consistent_with_deprecated_files() {
+		// When current major version is the initial version, we shouldn't load any deprecated functions file.
+		$this->assertTrue( false !== bp_get_deprecated_functions_versions(), 'Please check the list of `$deprecated_functions_versions` in `bp_get_deprecated_functions_versions()`. There should be one for each file of the `/src/bp-core/deprecated` directory.' );
```

Comment:
I think now would be a good time to add more test coverage. Like testing this behavior, https://github.com/buddypress/buddypress/blob/8cfca1802adeeb85fff69b8848deec6996d49e39/src/bp-core/bp-core-functions.php#L4975-L4977, and others.

Reply:
I can add those tests separately too. If necessary.

Reply:
That would be great 💪